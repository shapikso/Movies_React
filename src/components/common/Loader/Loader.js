import './Loader.scss';
import React from 'react';

const Loader = () => (
    <div className="loader">
        <div className="lds-spinner">
            {new Array(12).fill(null).map((index) => <div key={index}/>)}            
        </div>
    </div>
);


export default Loader;
