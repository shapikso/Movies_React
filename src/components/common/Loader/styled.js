import styled, {keyframes} from 'styled-components';

const ldsEllipsis1 = keyframes`
  0% {
    transform: scale(0);
  }
  100% {
    transform: scale(1);
  }
`;

const ldsEllipsis3 = keyframes`
  0% {
    transform: scale(1);
  }
  100% {
    transform: scale(0);
  }
`;

const ldsEllipsis2 = keyframes`
  0% {
    transform: translate(0, 0);
  }
  100% {
    transform: translate(24px, 0);
  }
`;

export const StyledMainDiv = styled.div`
  width: 100%;
  height: 100%;
  background: rgba(0, 0, 0, 0.7);;
  position: absolute;
  top:0;
  left:0;
  z-index: 1;
  color: white;
`;

export const StyledSecondDiv = styled.div`
  //color: official;
  display: inline-block;
  position: relative;
  width: 80px;
  height: 80px;
  top: 50%;
  left: ${({btn}) => btn ? `15%` : `50%`};
  transform: translate(-50%, -50%);
  `;

export const StyledThirdDiv = styled.div`
  position: absolute;
  top: 33px;
  width: 13px;
  height: 13px;
  border-radius: 50%;
  background: #fff;
  animation-timing-function: cubic-bezier(0, 1, 1, 0);
  :nth-child(1) {
    left: 8px;
    animation: ${ldsEllipsis1} 0.6s infinite;
  }

  :nth-child(2) {
    left: 8px;
    animation: ${ldsEllipsis2} 0.6s infinite;
  }

  :nth-child(3) {
    left: 32px;
    animation: ${ldsEllipsis2} 0.6s infinite;
  }

  :nth-child(4) {
    left: 56px;
    animation: ${ldsEllipsis3} 0.6s infinite;
  }
`;