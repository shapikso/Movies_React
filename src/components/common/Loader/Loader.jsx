import React from 'react';
import {StyledSecondDiv, StyledThirdDiv} from './styled';

const Loader = () => (
    <StyledSecondDiv>
        {new Array(4).fill(null).map((_, index) => <StyledThirdDiv key={index}/>)}
    </StyledSecondDiv>
);

export default Loader;
