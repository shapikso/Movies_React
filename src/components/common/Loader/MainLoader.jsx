import React from 'react';
import {StyledMainDiv} from './styled';
import Loader from "./Loader";

const MainLoader = () => (
    <StyledMainDiv>
        <Loader />
    </StyledMainDiv>
);

export default MainLoader;
