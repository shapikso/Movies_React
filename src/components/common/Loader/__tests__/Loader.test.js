import React from 'react';
import {shallow, mount} from 'enzyme';
import Loader from "../Loader";

describe('Loader test', () => {

    it('should render correctly', () => {
        const component = shallow(<Loader />);
        expect(component).toMatchSnapshot();
    });
    it('should render 4 div', () => {
        const component = mount(<Loader/>);
        expect(component.find('div div')).toHaveLength(4);
    });
});