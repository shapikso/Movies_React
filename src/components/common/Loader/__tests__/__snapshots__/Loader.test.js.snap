// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OverviewDetails should render correctly 1`] = `ShallowWrapper {}`;
